### PR TITLE
DEPR: Move NumericIndex doc string + some minor stuff

### PR DESCRIPTION
**Pull request by topper-123**
Moves doc string  of `NumericIndex` to `Index`. Also moves `_can_hold_strings` attribute and removes  `NumericIndex._default_dtype`, & `NumericIndex._typ`.

xref #42717.

### CODE REVIEW

**Review by jbrockmendel — APPROVED** [review]
LGTM

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -324,6 +330,9 @@ class Index(IndexOpsMixin, PandasObject):
 
     >>> pd.Index(list('abc'))
     Index(['a', 'b', 'c'], dtype='object')
+
+    >>> pd.Index([1, 2, 3], dtype="uint8")
+    NumericIndex([1, 2, 3], dtype='uint8')
```
Comment (phofl) [inline pandas/core/indexes/base.py:335]: This has to be updated later? (when NumericIndex is gone?)

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @topper-123

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (topper-123) [reply pandas/core/indexes/base.py:335]: Yes, I`ll do a ctrl-f to clean up any cruft left at the end.

**Pull request merged**